src/pudl/extract/epacems.py

```diff
@@ -100,20 +100,24 @@
 class EpaCemsPartition(NamedTuple):
     """Represents EpaCems partition identifying unique resource file."""
 
-    year: str
-    quarter: str
+    year_quarter: str
 
     def get_key(self):
         """Returns hashable key for use with EpaCemsDatastore."""
-        return (self.year, self.quarter)
+        return self.year_quarter
+
+    @property
+    def year(self):
+        """Returns the year associated with this year_quarter partion."""
+        return pd.to_datetime(self.year_quarter).year
```

Comment:
yea i thought i was going to use the year property in more contexts but it turned out not to really be necessary. and I didn't make a quarter property because ideally we will be able to make a new cems archive soon with the file name more closely matching the partition so we can just do this:

```
    def get_quarterly_file(self: Self) -> Path:
        """Return the name of the CSV file that holds annual hourly data."""
        return Path(f"epacems-{self.year_quarter}.csv")
```

Reply:
also doesn't  this`20\d{2}` indicate the year can only be 2000 and on?

Reply:
Whoops. Yes.
```py
    year_quarter: Annotated[str, StringConstraints(strict=True, pattern=r"^(19|20)\d{3}[qQ][1-4]$")]
```

Reply:
so i am using the year property in other places. 

i still think it is too much to add the quarter as an attribute. but i will do the extraction of the quarter in the same way (w/ `to_datetime`) to make it more consistent.